stingray/tests/test_crossspectrum.py

```diff
@@ -394,6 +394,20 @@ def iter_lc(lc, n):
         cs = AveragedCrossspectrum(iter_lc(self.lc1, 1), iter_lc(self.lc2, 1),
                                    segment_size=1)
 
+
+    def test_with_multiple_lightcurves_variable_length(self):
+        gti = [[0, 0.05], [0.05, 0.5], [0.555, 1.0]]
+        lc1 = copy.copy(self.lc1)
```

Comment:
Note that `copy.copy` creates a shallow copy, so the arrays composing `lc1` will not be copied but linked. For a deep copy, there is another method, `copy.deepcopy`
https://docs.python.org/3/library/copy.html

Reply:
Fixed.